File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/data/mapper/CommentMapper.java
package com.example.postscommentsapp.data.mapper;

import com.example.postscommentsapp.data.local.entity.CommentEntity;
import com.example.postscommentsapp.domain.model.Comment;

import java.util.ArrayList;
import java.util.List;

public class CommentMapper {

    public static Comment toDomain(CommentEntity entity) {
        return new Comment(
                entity.getId(),
                entity.getPostId(),
                entity.getComment(),
                entity.getCreatedAt()
        );
    }

    public static CommentEntity toEntity(Comment comment) {
        return new CommentEntity(
                comment.getId(),
                comment.getPostId(),
                comment.getComment(),
                comment.getCreatedAt()
        );
    }

    public static List<Comment> toDomainList(List<CommentEntity> list) {
        List<Comment> result = new ArrayList<>();
        for (CommentEntity e : list) {
            result.add(toDomain(e));
        }
        return result;
    }
}

File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/data/mapper/CommentMapper.kt
package com.example.postscommentsapp.data.mapper

import com.example.postscommentsapp.data.local.entity.CommentEntity
import com.example.postscommentsapp.domain.model.Comment

object CommentMapper {

    fun toDomain(entity: CommentEntity): Comment =
        Comment(
            id = entity.id,
            postId = entity.postId,
            comment = entity.comment,
            createdAt = entity.createdAt
        )

    fun toEntity(domain: Comment): CommentEntity =
        CommentEntity(
            domain.id,
            domain.postId,
            domain.comment,
            domain.createdAt
        )

    fun toDomainList(list: List<CommentEntity>): List<Comment> =
        list.map { toDomain(it) }
}
